### PR TITLE
Spanish Localization

### PR DESCRIPTION
**Pull request by oco-odoo**
https://www.odoo.com/web#id=30669&view_type=form&model=project.task&action=333&active_id=967&menu_id=4720

Part 1 of Spanish localization.

It includes (distributed on enterprise and community):
- the 3 CoA 
- the Spanish balance sheet for each of these CoA
- taxes definition
- a Python engine for tax reports with pre-editable fields (called here AEAT reports)
- AEAT report for mod 111, 115 and 303

### CODE REVIEW

File: addons/l10n_es/data/account_account_template_common_data.xml [hunk 1]
```diff
@@ -1,4143 +1,4121 @@
...
-        <record id="pgc_2911_child" model="account.account.template">
+        <record id="l10n_es.account_common_2911" model="account.account.template">
             <field name="code">2911</field>
             <field name="reconcile" eval="False"/>
             <field name="name">Deterioro de valor de construcciones</field>
             <field name="user_type_id" ref="account.data_account_type_fixed_assets"/>
             <field name="chart_template_id" ref="l10n_es.account_chart_template_common"/>
         </record>
-        <record id="pgc_2912_child" model="account.account.template">
+        <record id="l10n_es.account_common_2912" model="account.account.template">
```
Comment (pedrobaeza) [inline addons/l10n_es/data/account_account_template_common_data.xml]: Why removing these accounts?

File: addons/l10n_es/data/account_account_template_full_data.xml [hunk 2]
```diff
@@ -0,0 +1,952 @@
...
+        <field name="chart_template_id" ref="account_chart_template_full"/>
+    </record>
+    <record id="l10n_es.account_full_5297" model="account.account.template">
+        <field name="code">5297</field>
+        <field name="reconcile" eval="True"/>
+        <field name="name">Provisión a corto plazo por transacciones con pagos basados en instrumentos</field>
+        <field name="user_type_id" ref="account.data_account_type_current_liabilities"/>
+        <field name="chart_template_id" ref="account_chart_template_full"/>
+    </record>
+    <record id="l10n_es.account_full_5530" model="account.account.template">
```
Comment (pedrobaeza) [inline addons/l10n_es/data/account_account_template_full_data.xml]: Missing `551- Cuenta corriente con socios y administradores`

File: addons/l10n_es/data/account_account_template_pymes_data.xml [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        <field name="chart_template_id" ref="l10n_es.account_chart_template_pymes"/>
+    </record>
+    <record id="pgc_pyme_490" model="account.account.template">
+        <field name="code">490</field>
+        <field name="reconcile" eval="False"/>
+        <field name="name">Deterioro de valor de créditos por operaciones comerciales</field>
+        <field name="user_type_id" ref="account.data_account_type_current_liabilities"/>
+        <field name="chart_template_id" ref="l10n_es.account_chart_template_pymes"/>
+    </record>
+    <record id="l10n_es.account_pymes_5935" model="account.account.template">
```
Comment (pedrobaeza) [inline addons/l10n_es/data/account_account_template_pymes_data.xml]: Missing `551- Cuenta corriente con socios y administradores`

File: addons/l10n_es/data/account_data.xml [hunk 4]
```diff
@@ -6,13 +6,13 @@
         <record id="tax_group_iva_0" model="account.tax.group">
             <field name="name">IVA 0%</field>
         </record>
-        <record id="tax_group_iva_05" model="account.tax.group">
+        <record id="tax_group_iva_0-5" model="account.tax.group">
             <field name="name">IVA 0.5%</field>
```
Comment (pedrobaeza) [inline addons/l10n_es/data/account_data.xml]: Some of these taxes are incorrectly tagged as `IVA`. This is for example `Recargo de equivalencia`, and others (like 19% is `Retenciones`)

Comment (oco-odoo) [reply addons/l10n_es/data/account_data.xml]: Ok, I'll change tax_group_iva_0-5's name to Recargo de equivalencia, and correct the name for 19% as well. Are there others ? If so, could you give me a complete list of the changes to do ?

File: addons/l10n_es/data/account_account_template_common_data.xml (hunk 1 above)
Comment (oco-odoo) [reply addons/l10n_es/data/account_account_template_common_data.xml]: They haven't been removed, but renamed, for clarity and soundness with the "new" accounts defined for PYMES, "assoc" and "full" CoA. Naming them so allows knowing where they come from with just their xml id.

File: addons/l10n_es/data/account_data.xml (hunk 4 above)
Comment (pedrobaeza) [reply addons/l10n_es/data/account_data.xml]: You can easily deduce them by the tax name they have. Example:

https://github.com/odoo/odoo/blob/65f5fc6bf8442fc7004d0fc8e006184359f0fe4e/addons/l10n_es/data/account_tax_data.xml#L586

or 

https://github.com/odoo/odoo/blob/65f5fc6bf8442fc7004d0fc8e006184359f0fe4e/addons/l10n_es/data/account_tax_data.xml#L250

**Comment by oco-odoo** [comment]
@pedrobaeza  I fixed what you pointed out. Feel free to have a look :)

**Comment by pedrobaeza** [comment]
Please solve conflicts before.

**Comment by oco-odoo** [comment]
@pedrobaeza I just rebased.

File: addons/l10n_es/__manifest__.py [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
-# Copyright (c) 2008-2010 Zikzakmedia S.L. (http://zikzakmedia.com) All Rights Reserved.
-#                         Jordi Esteve <jesteve@zikzakmedia.com>
-# Copyright (c) 2012-2013, Grupo OPENTIA (<http://opentia.com>) Registered EU Trademark.
-#                         Dpto. Consultoría <consultoria@opentia.es>
-# Copyright (c) 2013 Serv. Tecnol. Avanzados (http://www.serviciosbaeza.com)
-#                    Pedro Manuel Baeza <pedro.baeza@serviciosbaeza.com>
+# List of contributors:
+# Jordi Esteve <jesteve@zikzakmedia.com>
+# Dpto. Consultoría Grupo Opentia <consultoria@opentia.es>
+# Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
```
Comment (pedrobaeza) [inline addons/l10n_es/__manifest__.py]: Can you please put my new email/brand? `pedro.baeza@tecnativa.com`

**Comment by pedrobaeza** [comment]
Hi, Olivier, thanks for the changes.

Trying on runbot, I see several issues, but not directly related with Spanish localization. I pointed them here, so you can redirect them, but please tell me if I have to do it via a new issue or any other PR:

* There's no account.account menu entry for managing them.
* Default taxes are not changed on the onchange event of the template chart, so they remain the same as previously, provoking inconsistencies: first they are empty. I have selected the default for one of the chart accounts. If now I install another chart of accounts in another company, the default taxes for the previous selection remains, and they might not be present on the new chart of accounts.
* Once installed the chart of accounts, I still see the package selector, and I see that this is for allowing to change the plan for the main company as PYME is auto-installed. But there's a side effect: you see in the dropdown all the possible account charts (France, Belgium, England...), so if you select one of the other localizations, there will be an ugly error. Isn't better to just not install any default account chart for Spanish companies as we do in OCA/l10n-spain?
* Isn't there now a transfer account in the wizard? I have seen the data in XML files, but there's no possibility of selecting it on the wizard. Have you default to it silently?

**Comment by anajuaristi** [comment]
Hi everybody:

I would like just to know about when is expected this PR to be merged. We would need it to test another  modules dependant on l10n_es to start testing master/V11. We could not do it right way until this is merged.

Thank you very much:
Ana

**Comment by qdp-odoo** [comment]
if everything goes well: today -_- although it has currently some flaws in the reports data that still need to be fixed (and could be fixed after the merge)

**Comment by pedrobaeza** [comment]
Quentin, please take into account my comments in https://github.com/odoo/odoo/pull/19154#issuecomment-327732225

**Comment by anajuaristi** [comment]
@qdp-odoo great!!

**Comment by qdp-odoo** [comment]
@pedrobaeza 

> There's no account.account menu entry for managing them.

Yes, but it's in enterprise only since the full accounting is not anymore in community. It's located in Accounting\Configuration\Accounting\Chart of account

>Default taxes are not changed on the onchange event of the template chart, so they remain the same as previously, provoking inconsistencies: first they are empty. I have selected the default for one of the chart accounts. If now I install another chart of accounts in another company, the default taxes for the previous selection remains, and they might not be present on the new chart of accounts.

I'll check that
>Once installed the chart of accounts, I still see the package selector, and I see that this is for allowing to change the plan for the main company as PYME is auto-installed. But there's a side effect: you see in the dropdown all the possible account charts (France, Belgium, England...), so if you select one of the other localizations, there will be an ugly error. Isn't better to just not install any default account chart for Spanish companies as we do in OCA/l10n-spain?

So what? you installed l10n_es + l10n_fr, Odoo generated the accounts from one template (following the installation order it will be either l10n_es.PYMES or the french one) but you're still able to switch to another while you don't make a journal entry. I find it neat like that.

>Isn't there now a transfer account in the wizard? I have seen the data in XML files, but there's no possibility of selecting it on the wizard. Have you default to it silently?

I don't even remember where the wizard is and how to call it. It was (and IS still), for sure in the settings.

**Comment by qdp-odoo** [comment]
landed in master, for saas-18, at revisions:
community: https://github.com/odoo/odoo/commit/3e1d46dcc90fb09f24de49783f0f1c1d50058186
enterprise: https://github.com/odoo/enterprise/commit/18c0ad3ad4970e5508e214c968b2901c93253a29

**Pull request closed without merging**